src/promptflow-tools/promptflow/tools/exception.py

```diff
@@ -58,6 +58,18 @@ def to_openai_error_message(e: Exception) -> str:
               "please make sure you have proper role assignment on your azure openai resource. You can refer to " \
               "https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/role-based-access-control"
         return f"OpenAI API hits {ex_type}: {msg}"
+    elif "messages with role 'tool' must be a response" in error_message:
```

Comment:
'tool' must be a response to a preceeding message with 'tool_calls'

Reply:
offiline synced, "preceeding" is a typo, might not be good to catch this typo as a message hint